src/gg.regression.unity.bots/Runtime/Scripts/RGBots.cs

```diff
@@ -0,0 +1,95 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using RegressionGames;
+using UnityEngine;
+
+public static class RGBots
+{
+    private static IRGBotEntry FindBot(string botName)
+    {
+        var botList = Resources.Load<IRGBotList>("RGBotList");
+
+        var botEntry = botList.botEntries.FirstOrDefault(b => b.botName == botName);
+        if (botEntry != null)
+        {
+            return botEntry;
+        }
+        return null;
+    }
+
+    private static IRGBotEntry FindBotByType<T>() where T : Component
+    {
+        var assemblyQualifiedName = typeof(T).AssemblyQualifiedName;
+        var botList = Resources.Load<IRGBotList>("RGBotList");
+
+        var botEntry = botList.botEntries.FirstOrDefault(b => b.qualifiedName == assemblyQualifiedName);
+        if (botEntry != null)
+        {
+            return botEntry;
+        }
+        return null;
+    }
+
+    public static GameObject SpawnBot<T>() where T : Component
+    {
+        GameObject botGameObject = new GameObject(typeof(T).Name);
+        var botEntry = FindBotByType<T>();
+        var behaviorName = botEntry?.botName ?? "Empty";
+        if (botEntry != null)
+        {
+            botGameObject.AddComponent<T>();
+        }
+        LinkToOverlay(botGameObject, behaviorName);
+        return botGameObject;
+    }
+
+    public static GameObject SpawnBot<T>(GameObject prefab) where T : Component
+    {
+        GameObject botGameObject = GameObject.Instantiate(prefab);
```

Comment:
We should name this so it isn't just 'New Game Object' in the scene

Reply:
👍

Reply:
This comment doesn't seem to be fixed in the current changes.

Reply:
Fixed. Current logic is:
1. Name of the bot will be the name of the prefab spawn
2. If no prefab, name of the bot will be name of the behavior selected
3. If no prefab and no behavior, uses `New Game Object`